simulariumio/file_converter.py

```diff
@@ -19,24 +21,207 @@
 class FileConverter(TrajectoryConverter):
     def __init__(self, input_file: InputFileData):
         """
-        This object loads the data in .simularium JSON format
-        at the input file
+        This object loads data from the input file in .simularium format.
 
         Parameters
         ----------
         input_file: InputFileData
-            A InputFileData object containing a string path
-            or string contents for the .simularium JSON file to load
+            A InputFileData object containing .simularium data to load
         """
-        print("Reading Simularium JSON -------------")
-        buffer_data = json.loads(input_file.get_contents())
+        try:
+            print("Reading Simularium JSON -------------")
+            buffer_data = json.loads(input_file.get_contents())
+        except UnicodeDecodeError:
+            print("Reading Simularium binary -------------")
+            buffer_data = FileConverter._load_binary(input_file)
         if (
             int(buffer_data["trajectoryInfo"]["version"])
             < CURRENT_VERSION.TRAJECTORY_INFO
         ):
-            buffer_data = self.update_trajectory_info_version(buffer_data)
+            buffer_data = FileConverter.update_trajectory_info_version(buffer_data)
         self._data = TrajectoryData.from_buffer_data(buffer_data)
 
+    @staticmethod
+    def _read_binary_file(input_file: InputFileData) -> Dict[str, Any]:
```

Comment:
something about `_read_binary_file` vs `_load_binary`,  one of the two needs renaming because they sound the same.  Thinking out loud:  Maybe this one could be called `_get_data_views_from_file`.  or _read_file_into_memory_buffers

Reply:
renamed to `_binary_data_from_file()`